server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1148,15 +1148,16 @@ public Host cancelMaintenance(final CancelMaintenanceCmd cmd) {
     }
 
     @Override
-    public Host reconnectHost(final ReconnectHostCmd cmd) {
-        final Long hostId = cmd.getId();
+    public Host reconnectHost(ReconnectHostCmd cmd) throws CloudRuntimeException, AgentUnavailableException{
```

Comment:
@rodrigo93 

 The scope of this pr is to improve the exception handling and error messaging. Changing the method signature is not a immediate concern. I will make the suggested changes in another pr if i find time.